AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java

```diff
@@ -271,28 +293,64 @@ private void createDeviceManager() {
             Log.d(LOG_TAG, "Device Manager created");
+            availableCameras = new HashMap<>();
 
-            List<VideoDeviceInfo> cameras;
-            boolean cameraFound = false;
-            while (!cameraFound) {
-                cameras = deviceManager.getCameras();
-                for (final VideoDeviceInfo camera: cameras) {
-                    final String cameraFacingName = camera.getCameraFacing().name();
-
-                    if (cameraFacingName.equalsIgnoreCase("front")) {
-                        Log.i(LOG_TAG, "Desired Camera selected");
-                        initializeCameraCompletableFuture.complete(camera);
-                        cameraFound = true;
-                        break;
-                    }
-                }
-            }
+            final List<VideoDeviceInfo> initialCameras = deviceManager.getCameras();
+            addVideoDevices(initialCameras);
+            initializeFrontCameraIfRequired();
+
+            final VideoDevicesUpdatedListener videoDevicesUpdatedListener = videoDevicesUpdatedEvent -> {
+                updateVideoDevices(videoDevicesUpdatedEvent);
+                initializeFrontCameraIfRequired();
+            };
+            deviceManager.addOnCamerasUpdatedListener(videoDevicesUpdatedListener);
         });
     }
 
+    private void initializeFrontCameraIfRequired() {
+        if (!initializeCameraCompletableFuture.isDone()) {
+            final VideoDeviceInfo initialCamera = getFrontCamera();
+            if (initialCamera != null) {
+                initializeCameraCompletableFuture.complete(initialCamera);
+            }
+        }
+    }
+
+    private void updateVideoDevices(final VideoDevicesUpdatedEvent videoDevicesUpdatedEvent) {
+        removeVideoDevices(videoDevicesUpdatedEvent.getRemovedVideoDevices());
+        addVideoDevices(videoDevicesUpdatedEvent.getAddedVideoDevices());
+    }
+
+    private void removeVideoDevices(final List<VideoDeviceInfo> removedVideoDevices) {
+        Log.d(LOG_TAG, "Removed Cameras: " + removedVideoDevices.size());
+        for (final VideoDeviceInfo removedVideoDevice: removedVideoDevices) {
+            final String cameraFacingName = removedVideoDevice.getCameraFacing().name();
+            availableCameras.remove(cameraFacingName);
+        }
+    }
+
+    private void addVideoDevices(final List<VideoDeviceInfo> addedVideoDevices) {
+        Log.d(LOG_TAG, "Added Cameras: " + addedVideoDevices.size());
+        for (final VideoDeviceInfo addedVideoDevice: addedVideoDevices) {
+            if (addedVideoDevice.getCameraFacing().name().equalsIgnoreCase(CameraType.FRONT.name())) {
+                availableCameras.put(CameraType.FRONT, addedVideoDevice);
+            } else if (addedVideoDevice.getCameraFacing().name().equalsIgnoreCase(CameraType.BACK.name())) {
+                availableCameras.put(CameraType.BACK, addedVideoDevice);
+            }
+        }
```

Comment:
is it possible to have multiple front or back cameras?

Reply:
Many phones nowadays have multiple front and back cameras.  I'm relying on the phone itself declaring an explicit `FRONT` or `BACK` camera for me to decide which would be the best ones to use as the 'official' front and back cameras.